Bottom: 4f457f176c3b02306c8021002e41ac8f1e9cdb3c
Top:    e7442ffb41b7a5013dfd2c0baa5dfba31010be8e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-05-11 15:35:26 +0100

Add comment explaining assumptions for poly_ContourInContour function


---

diff --git a/src/polygon1.c b/src/polygon1.c
index d9f6ce4..b325fb5 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2254,6 +2254,11 @@ poly_M_CheckInside (POLYAREA * p, Vector v0)
   return FALSE;
 }
 
+/* NB: This function assumes the caller _knows_ the contours do not
+ *     intersect. If the contours intersect, the result is undefined.
+ *     This function can return an incorrect (positive) result if the
+ *     contours share a common node at the arbitrary point tested.
+ */
 int
 poly_ContourInContour (PLINE * poly, PLINE * inner)
 {
